Firmware/Source/Controller/Controller.h

```diff
@@ -15,23 +15,35 @@ typedef enum __DeviceState
 	DS_Disabled = 2,
 	DS_Ready = 3,
 	DS_InProcess = 4,
+	DS_SelfTest = 5
 } DeviceState;
 
 typedef enum __DeviceSubState
 {
 	SS_None = 0,
-	SS_WaitAfterPulse = 1,
-	SS_PowerSupplyOn = 2,
-	SS_PowerSupplyOff = 3,
-	SS_StartPulse = 4,
-	SS_Pulse = 5
+	SS_PS_WaitingStart = 1,
+	SS_PS_WaitingOn = 2,
+	SS_PS_WaitingOff = 3,
+	SS_WaitPause= 4,
+	SS_StartPulse = 5,
+	SS_Pulse = 6,
+
+	SS_ST_Sync = 10,
+	SS_ST_StartPulse = 11,
+	SS_ST_Pulse = 12,
+	SS_ST_WaitPause = 13,
+	SS_ST_PulseCheck = 14
```

Comment:
```suggestion
	SS_ST_Sync = 10,
	SS_ST_StartPulse,
	SS_ST_Pulse,
	SS_ST_WaitPause,
	SS_ST_PulseCheck
```
В TrueStudio для enum'ов удобно указывать номер только первой записи в группе. Остальные автоматом нумеруются и их значение подсвечивается при наведении курсора. Удобно при добавлении новых состояний, чтобы всё не перенумеровывать и при этом числа отражали последовательность этих состояний.

Reply:
Поправил